src/libraries/Common/src/System/Security/Cryptography/MLKem.cs

```diff
@@ -1801,32 +1815,26 @@ private TResult ExportPkcs8PrivateKeyCallback<TResult>(ExportPkcs8PrivateKeyFunc
             // Decapsulation keys are always larger than the seed, so if we end up with a seed export it should
             // fit in the initial buffer.
             int size = Algorithm.DecapsulationKeySizeInBytes + 32;
-            byte[] buffer = ArrayPool<byte>.Shared.Rent(size); // Released to callers, do not use CryptoPool.
+            byte[] buffer = CryptoPool.Rent(size); // Only passed out as span, callees can't keep a reference to it
             int written;
 
             while (!TryExportPkcs8PrivateKeyCore(buffer, out written))
             {
-                ClearAndReturnToPool(buffer, written);
+                CryptoPool.Return(buffer);
                 size = checked(size * 2);
                 buffer = ArrayPool<byte>.Shared.Rent(size);
             }
 
-            if (written > buffer.Length)
+            if (written < 0 || written > buffer.Length)
```

Comment:
Is `if ((uint)written > buffer.Length)` avoided here for readability?

Reply:
Yeah, I don't really care for things like that until we are on some kind of hot path, and importing PKCS#8 is not a hot path. Idiomatic code seems better here, to me at least.